freva-rest/src/databrowser_api/core.py

```diff
@@ -887,3 +1071,426 @@ async def zarr_response(
+        await self._delete_from_mongo(search_keys)
+
+    async def _validating_userdata_paths(
+        self, *paths: Union[str, os.PathLike[str]]
+    ) -> List[os.PathLike[str]]:
+        """
+        Validate the user data input paths by checking if each path exists.
+
+        Parameters
+        ~~~~~~~~~~
+        *paths : os.PathLike
+            One or more paths to validate.
+
+        Returns
+        ~~~~~~~~~~
+        List[os.PathLike]
+            A list of valid paths that exist.
+        """
+        validated_paths: List[os.PathLike[str]] = []
+        for path in paths:
+            if not Path(path).exists():
+                logger.warning(f"The path {str(path)} does not exist")
+            else:
+                validated_paths.append(cast(os.PathLike[str], path))
+
+        if not validated_paths:
+            raise FileNotFoundError("No valid paths found")
+
+        return validated_paths
+
+    def _timedelta_to_cmor_frequency(self, dt: float) -> str:
+        """
+        Convert a time delta to a CMOR-compliant frequency string.
+
+        Parameters
+        ~~~~~~~~~~
+        dt : float
+            Time delta in seconds.
+
+        Returns
+        ~~~~~~~
+        str
+            The CMOR frequency corresponding to the time delta.
+        """
+        for total_seconds, frequency in self.time_table.items():
+            if dt >= total_seconds:
+                return frequency
+        return "fx"  # pragma: no cover
+
+    @property
+    def time_table(self) -> dict[int, str]:
+        """
+        Provide a mapping from time intervals (in seconds) to CMOR frequency strings.
+
+        Returns
+        ~~~~~~~
+        Dict[int, str]
+            A dictionary mapping time intervals (seconds) to CMOR frequencies.
+        """
+        return {
+            315360000: "dec",  # Decade
+            31104000: "yr",  # Year
+            2538000: "mon",  # Month
+            1296000: "sem",  # Seasonal (half-year)
+            84600: "day",  # Day
+            21600: "6h",  # Six-hourly
+            10800: "3h",  # Three-hourly
+            3600: "hr",  # Hourly
+            1: "subhr",  # Sub-hourly
+        }
+
+    def get_time_frequency(self, time_delta: int, freq_attr: str = "") -> str:
+        """
+        Determine the CMOR-compliant time frequency based on the time delta
+        and/or frequency attribute.
+
+        Parameters
+        ~~~~~~~~~~
+        time_delta : int
+            The time delta in seconds between consecutive time steps.
+        freq_attr : str, optional
+            The time frequency attribute that might already exist in
+            the data, by default "".
+
+        Returns
+        ~~~~~~~
+        str
+            The CMOR-compliant time frequency.
+        """
+        if freq_attr in self.time_table.values():
+            return freq_attr
+        return self._timedelta_to_cmor_frequency(time_delta)
+
+    async def _get_metadata(self, file_name: os.PathLike[str]
```

Comment:
This is not the right place for this method. I think we should add this to the client. I know it would be nice if this would work but you'd need to guarantee that that server has access to the data. In our case we would have to have the entire lustre mounted. Also the server would have to have read access to the data.

I think you want to place this method to the client.